superx/information_extractor.py

```diff
@@ -0,0 +1,201 @@
+                        zip_links.add(https)
+
+            self.update_super_name(url)
+            self.info_parser(zip_links)
+
+    def info_parser(self, zip_links):
+        """
+        This method retrieves the xml file in the gzip file and parses it into an xml tree
+        The branch_id is retrieved from the xml file and then it is sent for retrieval of the item information
+        :param zip_links: list of zip file links from the website
+        """
+        for zip_link in zip_links:
+            # corrects the urls
+            if self.super_name == 'mega':
+                zip_link = self.url_list[0] + '/' + zip_link
+            elif self.super_name == 'victory':
+                zip_link = 'http://matrixcatalog.co.il/' + zip_link
+
+            request = requests.get(zip_link)
+            content = request.content
+            xml_file = gzip.decompress(content).decode('utf-8')
+            # parses the xml document into a tree
+            tree = ET.fromstring(xml_file)
+            # gets the branch/store ID of current xml file
+            branch_id = self.get_branch_id(tree.getchildren())
+            # gets child containing item information
+            items = tree.getchildren()[-1]
+            self.extract_information(items, branch_id)
+
+    def extract_information(self, items, current_branch_id):
+        """
+        This method iterates over all items in the supermarket and extracts the relevant data
+        The data is then committed into the relevant table in the data base
+        :param items: The child of the parsed xml tree containing all item information
+        :param current_branch_id: id of current branch
+        """
+        products_list = []
+        branch_price_list = []
+        item = 'Item'
+        bIsWeighted = 'bIsWeighted'
+
+        # different supermarkets use different titles
+        if self.super_name == 'victory':
+            item = 'Product'
+            bIsWeighted = 'BisWeighted'
+
+        for item in items.findall(item):
+            code = int(item.find('ItemCode').text)
+            name = item.find('ItemName').text
+            quantity = item.find('Quantity').text
+            price = item.find('ItemPrice').text
+            update_date = self.standardize_date(item.find('PriceUpdateDate').text)
+            is_weighted = False
+            if item.find(bIsWeighted).text == 1:
+                is_weighted = True
+
+            unit_of_measure = 'none'
+            if is_weighted:
+                unit_of_measure = self.convert_unit_name(item.find('UnitQty').text)
+
+            products_list.append(Product(id=code, name=name, quantity=quantity, is_weighted=is_weighted, unit_of_measure=unit_of_measure))
+            branch_price_list.append(BranchPrice(item_code=code, branch_id=current_branch_id, price=price, update_date=update_date))
+
+        # add all the product info to the db and commit
+        db.session.add_all(products_list)
+        db.session.commit()
```

Comment:
you probably want to use only the commit below as you want all this data to go in together